src/poseidon/poseidon-opt.ts

```diff
@@ -65,5 +67,50 @@ export class Poseidon {
 
     return F.normalize(state[0]);
   }
+
+  hashBytes(msg: Uint8Array): bigint {
+    let inputs: Array<bigint> = [];
+    for (let i = 0; i < SPONGE_INPUTS; i += 1) {
+      inputs.push(BigInt(0));
+    }
+    let dirty = false;
+    let hash: BigInt;
+
+    let k = 0;
+    for (let i = 0; i < parseInt(`${msg.length / SPONGE_CHUNK_SIZE}`); i += 1) {
+      dirty = true;
+      inputs[k] = utils.beBuff2int(msg.slice(SPONGE_CHUNK_SIZE * i, SPONGE_CHUNK_SIZE * (i + 1)));
+      if (k === SPONGE_INPUTS - 1) {
+        hash = this.hash(inputs);
+        dirty = false;
+        inputs = [];
+        inputs[0] = hash.valueOf();
```

Comment:
I guess the more elegant solution, what do you think?
```
inputs = new Array(SPONGE_INPUTS).fill(BigInt(0));
inputs[0] = inputs[0] = hash.valueOf();
```

Reply:
Great! Just saw in my code that inputs are already initialized to and hence `input = [] ` is redundant.

Fixed in [this](https://github.com/iden3/js-crypto/pull/2/commits/5cc7f47b45c5d8f19ab6092311b9bcb34e30642a) commit.

Reply:
please replace also
```
for (let j = 1; j < SPONGE_INPUTS; j += 1) {
          inputs[j] = BigInt(0);
 }